### DISCUSSION TITLE
FPS drop using GStreamer

### DISCUSSION THREAD

**Discussion by AdrienPapin** [OvenMediaLabs/OvenMediaEngine#1742] [Help] [answered]
Hello, I’m new to OvenMediaEngine.
I’m trying to stream a RTMP output using GStreamer (gst.h c++) and a RTSP input.

I use gst_parse_launch on this pipeline :
```
const gchar* pipelineDesc = "
rtspsrc location=rtsp://input is-live=true latency=0 protocols=tcp ! 
decodebin ! x264enc ! flvmux streamable=true ! 
rtmpsink location=rtmp://output";
```

I got no problem during the parse_lauch but on the player I get very few frames
Can someone explain to me if I did something wrong with the pipeline or if it comes from somewhere else ?

**Comment by naanlizard** [comment]
To be clear, you are using a setup like this

RTSP source ->Gstreamer-RTMP-output-from-Gstreamer->OME

How are you playing the stream from OME? If WebRTC, sounds like you have B
frames, check the docs

**Reply by antonymarion** [reply]
@naanlizard I have a similar issue with gstreamer and ome


Do you know if it is possible to get rid of b frames with gstreamer by transcoding the stream on the fly ?

**Comment by naanlizard** [comment]
No idea how gstreamer works, but it's preferable to remove the b frames at
the original encoding if you can

**Comment by bchah** [comment, marked as answer]
I think you can set flags on x264enc, e.g.

```
const gchar* pipelineDesc = "
rtspsrc location=rtsp://input is-live=true latency=0 protocols=tcp ! 
decodebin ! x264enc bframes=0 key-int-max=60 ! flvmux streamable=true ! 
rtmpsink location=rtmp://output";
```

**Reply by antonymarion** [reply]
Thks a lot @bchah

**Reply by AdrienPapin** [reply]
Thx for the fix :)
